python/feldera/pipeline.py

```diff
@@ -347,18 +411,22 @@ def shutdown(self):
                 # block until the callback runner has been stopped
                 queue.join()
 
-        self.client.shutdown_pipeline(self.name)
+        self.client.shutdown_pipeline(self.name, timeout_s=timeout_s)
 
-    def resume(self):
+    def resume(self, timeout_s: Optional[float] = None):
         """
-        Resumes the pipeline.
+        Resumes the pipeline from the PAUSED state.
+
+        :param timeout_s: The maximum time (in seconds) to wait for the pipeline to shut down.
         """
 
-        self.client.start_pipeline(self.name)
+        self.client.start_pipeline(self.name, timeout_s=timeout_s)
 
     def delete(self):
         """
         Deletes the pipeline.
+
+        The pipeline must be shutdown before it can be deleted.
```

Comment:
I think this is sufficiently improved for now, it would be good to keep this in mind and keep documenting things. When users want to build robust scripts, they might need to catch exceptions such as when a pipeline becomes `Unavailable`. The Python client should be written and documented with the aim of production use cases, not just controlled test use cases where it is unlikely the pipeline will exhibit less common behavior.